### ISSUE TITLE
trajectory related

### ISSUE THREAD

**Issue by yufengsjtu** [XueYifan1995/Real-time-NMPC-for-USVs-CasADi#3]
Hi,excuse me for a moment.can you tell me what kind of method you use to generate the trajectory ?

**Comment by XueYifan1995** [comment]
For example, in MATLAB:
T=[0:0.1:200];
x_ref=10*sin(0.02*T);
y_ref=20*cos(0.03*T);





---- Replied Message ----
From  ***@***.***> Date  4/20/2023 13:42 To  ***@***.***> Cc  ***@***.***> Subject  [XueYifan1995/Real-time-NMPC-for-USVs-CasADi] trajectory related (Issue #3)  

Hi,excuse me for a moment.can you tell me what kind of method you use to generate the trajectory ?

**Comment by yufengsjtu** [comment]
So does this control method have any requirements for the properties of the trajectory? For example, for curvature constraints on trajectories or unmanned ship state constraints on the starting and ending points of trajectories. Are these constraints reflected in the constraints of MPC?

**Comment by XueYifan1995** [comment]
The trajectory to be tracked needs to be reachable by the dynamic system (Reachability).
In this paper, the constraint of trajectory itself is not considered, but the cost function is used to realize trajectory tracking .






---- Replied Message ----
From  ***@***.***> Date  4/20/2023 14:02 To  ***@***.***> Cc  Xue ***@***.***> ,
 ***@***.***> Subject  Re: [XueYifan1995/Real-time-NMPC-for-USVs-CasADi] trajectory related (Issue #3)  

So does this control method have any requirements for the properties of the trajectory? For example, for curvature constraints on trajectories or unmanned ship state constraints on the starting and ending points of trajectories. Are these constraints reflected in the constraints of MPC?

**Comment by yufengsjtu** [comment]
Thank you for your clarification. Finally, did MPC consider the yaw angle control of unmanned ships？

**Comment by XueYifan1995** [comment]
Yes, the 3-DOF dynamic and kinematic model is used, including surge, sway and yaw. 





---- Replied Message ----
From  ***@***.***> Date  4/20/2023 14:13 To  ***@***.***> Cc  Xue ***@***.***> ,
 ***@***.***> Subject  Re: [XueYifan1995/Real-time-NMPC-for-USVs-CasADi] trajectory related (Issue #3)  

Thank you for your clarification. Finally, did MPC consider the yaw angle control of unmanned ships？

**Comment by yufengsjtu** [comment]
Thanks for your patient answer  again.
